.github/workflows/docker-build.yml

```
@@ -1,15 +1,51 @@
 name: Docker Image CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
 
-jobs:
+permissions:
+  contents: read
+  packages: write
 
+jobs:
   docker-build:
-
     runs-on: ubuntu-latest
-
     steps:
     - name: Checkout
       uses: actions/checkout@v4
-    - name: Build the Docker image
-      run: docker build . --file Dockerfile --tag spellcheck:$(date +%s)
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
```

Please add the SHA sum for the action version with the human readable version numbers as a appended comment as per the GitHub recommendations:

- https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

```suggestion
      uses: docker/setup-qemu-action@29109295f81e9208d7d86ff1c6c12d2833863392 # v3.6.0
```